### DISCUSSION TITLE
ESLint versions dependents support (Tracking: ESLint v9 support / Tracking: Flat Config support)

### DISCUSSION THREAD

**Discussion by Byron2016** [eslint/eslint#18773] [Ideas]
@nzakas  currently there are [24347 dependents](https://www.npmjs.com/package/eslint?activeTab=dependents)  of **ESLint**; how many of them support  **ESLint v9.9.0** ?

Could be a fantastic utility to have a form that  **ESLint** show this information.

| DEPENDENT               | VERSION     | LAST ESLINT VERSION SUPPORTED |
| ------------------------|--------------| ------------------------------------ |
| eslint-plugin-react     | 7.35.0          |  9.9.0                                             |
| @eslint/XXX               | Y.YY.Y         |  8.57.0                                           |

......

What do you think about?

**Comment by Byron2016** [comment]
@nzakas  @Tanujkanti4441 @ljharb @aladdin-add
Related to issue [18766](https://github.com/eslint/eslint/issues/18766) I have created this Idea [18773](https://github.com/eslint/eslint/discussions/18773) I think could be interesting to know your thinkings about.
I think that, have this **new utility** could : 

- help people to find fast answer about their question
- reduce among of questions that are raised like issues
- path people to right please where to ask about a problem.

maybe people could use this **new utility** in this way `npx @eslint/dependents-report` with a optional args array of specific dependents like this: `npx @eslint/dependents-report --filter [a, b, c, d]` and obtain this response:

| DEPENDENT               | VERSION     | LAST ESLINT VERSION SUPPORTED |
| ------------------------|--------------| ------------------------------------ |
| a    | 7.35.0          |  9.9.0                                             |
| b              | Y.YY.Y         |  8.57.0                                           |
| c              | Y.YY.Y         |  8.57.0                                           |
| d             | Y.YY.Y         |  9.9.0                                           |

Thanks for your ideas about it, and how usefull do you think it could be for comunity.

**Reply by nzakas** [reply]
It's an interesting idea, though I can't quite imagine what you're envisioning for how this would be implemented.

We don't have the bandwidth to take on another project right now, but if you're excited about this, I'd suggest that you build it. :smile:

**Reply by Byron2016** [reply]
**.... what you're envisioning for how this would be implemented.**
- Idea 1:
  - If is it possible to scan projects dependents' peer dependencies, dependencies, devdependencies across npm where all information is grouped,  and obtain table. and pushish it in some place
  - For this idea, currently I think I don´t have enought knowlage to do it by myselfe, I would like but I must be honest with myselft
- Idea 2:
  - Allow dependents' owners update ESLint's resume table to update it with that information.
    -  It is needed a place where to update information, can ESLint give us a place?
    -  next stepts could be thinking if last step will be confirmed.

**Reply by nzakas** [reply]
I think idea 1 is the only approach that would be useful. We already have two issues where we are manually tracking things:
https://github.com/eslint/eslint/issues/18391
https://github.com/eslint/eslint/issues/18093

**Reply by Byron2016** [reply]
I think these two **manually tracking** issues is a really good advance, maybe could be a good idea to add an utility that can show them with a command: `npx eslint --manual-tracking` withthis first step, could be people with more knoledge that me to be motived to make some advances and automatics filters.
What do you think?

**Reply by nzakas** [reply]
I don't think it makes sense to build this into ESLint.

**Reply by Byron2016** [reply]
By the way, there are a inconsistency between dependents numbers show by npm If you are interested, I raise a issue [source code for web page npmjs.com and dependents number inconsistency #1231](https://github.com/npm/documentation/issues/1231)

**Reply by aladdin-add** [reply]
I agree with @nzakas. This feature should not be part of the eslint core.

From a plugin maintainer's perspective, I added support information to the `eslint-plugin-n` documentation:
![image](https://github.com/user-attachments/assets/9147f0f0-bfab-491a-aced-51c498109e52)

maybe we could add a similar section in the eslint templates: https://github.com/eslint/generator-eslint
